backend/routers/pusher.py

```
@@ -190,3 +191,103 @@ async def websocket_endpoint_audio_bytes(
+            print(f'Could not process audio: error {e}')
+            websocket_close_code = 1011
+        finally:
+            websocket_active = False
+
+    # heart beat
+    async def send_heartbeat():
+        nonlocal websocket_active
+        nonlocal websocket_close_code
+        try:
+            while websocket_active:
+                await asyncio.sleep(20)
```

Ruff

`asyncio` may be undefined, or defined from star imports

(F405)